src/js/utils/ServiceUtil.js

```diff
@@ -63,8 +64,18 @@ const pruneHealthCheckAttributes = function (healthCheckSchema, healthCheck) {
   }, {});
 };
 
+function looksLikePod(data) {
+  // A pod definition contains a `containers` field, not found on the
+  // other definitions.
+  return Array.isArray(data.containers);
```

Comment:
Let's move this check to `createServiceFromDefinition` method, as it's only used once.